lib/tools.js

```diff
@@ -437,7 +437,11 @@ function stringify (obj) {
   try {
     return JSON.stringify(obj)
   } catch (_) {
-    return stringifySafe(obj)
+    try {
+      return stringifySafe(obj)
+    } catch (_) {
+      return '"[unable to serialize, circular reference is too complex to analyze]"'
```

Comment:
why this double wrap of error message?

Reply:
Because it can still throw. If even the "safe" options throws, there is something really odd in that object and we should give up, not blow up the error to the user.

Reply:
Sorry for not being clear. I meant th `'"[["'` wrap of message itself.

Reply:
It follows the same pattern of `[Circular]` used for circular references